### ISSUE TITLE
Feature Request to Support for Github Submodules in Prefect Projects

### ISSUE THREAD

**Issue by EmilRex** [PrefectHQ/prefect#9462]
### Discussed in https://github.com/PrefectHQ/prefect/discussions/9339

<div type='discussions-op-text'>

<sup>Originally posted by **akmukherjee** April 26, 2023</sup>
I have a feature request for Submodules initialization in Prefect Projects. Currently my prefect.yaml looks as shown below:

```
# Generic metadata about this project
name: icdc-dataloader
prefect-version: 2.10.4

# build section allows you to manage and build docker images
build: null

# push section allows you to manage if and how this project is uploaded to remote locations
push: null

# pull section allows you to provide instructions for cloning this project in remote locations
pull:
- prefect.projects.steps.git_clone_project:
    repository: https://github.com/CBIIT/icdc-dataloader
    branch: icdc-dl-prefectv1
    access_token: null
```
This set up does not pull the submodules associated with this repo. It would be nice to have a field `say: submodules: true` and then prefect deployment would run a `git submodule --init `with each deployment.
</div>

**Comment by EmilRex** [comment]
@akmukherjee @taylor-curran

**Comment by akmukherjee** [comment]
So, I updated the Prefect version to 2.10.8, set the `include_submodules: true` flag and retried [this ](https://github.com/akmukherjee/prefect-test/tree/main)flow and I can clearly see in the output that it is not cloning the submodule.
![image](https://github.com/PrefectHQ/prefect/assets/6787494/34189749-e4ca-43d5-8893-2a8a8d98e812)

**Comment by desertaxle** [comment]
Hey @akmukherjee, this functionality was released in 2.10.9. Could you see if this works after upgrading your `prefect` version?

**Comment by akmukherjee** [comment]
Here it see for you to see. I do not see any debug output which shows me that the submodule is getting cloned. I see that the original repo is cloned from the debug messages.
![image](https://github.com/PrefectHQ/prefect/assets/6787494/2752ea9c-28c8-4f32-990b-08854fc31024)
